src/api/routes/company.js

```diff
@@ -148,4 +148,26 @@ module.exports = (app) => {
                 return next(err);
             }
         });
+
+    /**
+     * Deletes a company and all its offers
+     */
+    router.put("/:companyId/delete",
+        or([
+            authMiddleware.isCompany,
+            authMiddleware.isGod
+        ], { status_code: HTTPStatus.UNAUTHORIZED, error_code: ErrorTypes.FORBIDDEN, msg: ValidationReasons.INSUFFICIENT_PERMISSIONS }),
+        validators.deleteCompany,
+        (req, res, next) => companyMiddleware.canManageAccountSettings(req.params.companyId)(req, res, next),
+        async (req, res, next) => {
+            try {
+                const service = new CompanyService();
+                await new OfferService().deleteOffersByCompanyId(req.params.companyId);
+                await service.sendCompanyDeletedNotification(req.params.companyId);
+                const company = await service.deleteCompany(req.params.companyId);
```

Comment:
This will send an email even if the company deletion fails for some reason. Should we be concerned about this?

I had to put it before because we fetch the company in the way we're sending the emails, which can't happen if it was deleted. This would require changing _sendCompanyNotification() so it is possible to pass it a company instead of a companyId

Reply:
Yes, we should be concerned.

_sendCompanyNotification is supposed to abstract the email sending to an existing company, changing it to receive the company kind of defeats its purpose. I suggest fetching the required info and sending the email without calling _sendCompanyNotification. (i.e., sendCompanyDeletedNotification should not call _sendCompanyNotification)

From looking at its code it reminded me that the Account of the Company being deleted should be deleted as well, to prevent logins.